test/integration/util.go

```diff
@@ -0,0 +1,53 @@
+// +build integration
+
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"io/ioutil"
+	"time"
+
+	"sigs.k8s.io/yaml"
+)
+
+func waitFor(ctx context.Context, interval time.Duration, check func(chan error)) error {
```

Comment:
These are temporary functions that we could likely move to `crossplane-runtime` in the future.. I will open an issue to track

Reply:
that sounds good.  and this waitFor function doesn't have a timeout itself, the timeout is at the top level when go test is invoked, right?

Reply:
Right! This seemed a little bit simpler to me because we are just saying how long we are okay with the entire scenario (i.e. a full test) running for, rather than specifying for each operation.